CONTRIBUTING.md

```diff
@@ -4,3 +4,32 @@ NetFoundry welcomes all and any contributions. All open source projects managed
 [guide for contributions](https://netfoundry.github.io/policies/CONTRIBUTING.html).
 
 If you are eager to contribute to a NetFoundry-managed open source project please read and act accordingly.
+
+## Project Styles
+
+This project uses several programming languages. Each language has its own style guide specifying any language-specific
+conventions and idioms.
+
+### Markdown
+
++ This project uses [GitHub Flavored Markdown](https://github.github.com/gfm/).
++ Wrap lines at 120 characters if the audience is expected to read the source.
++ Do not wrap lines if the audience is expected to read the rendered output as HTML.
++ Use [Markdownlint](https://github.com/DavidAnson/markdownlint) with [the configuration file](.markdownlint.yaml) committed to this repository to find formatting problems.
+
+### Go
+
++ This project uses [Go](https://golang.org/) language conventions.
++ Organize imports as a single block in alphabetical order without empty lines.
++ Begin log messages with a lowercase letter and do not end with punctuation.
++ Format log messages with format strings and arguments like 'tried a thing and failed: %s'.
```

Comment:
This needs to be expanded...

Log messages reporting errors should be formatted:
`logrus.Errorf("tried a thing and failed: %v", err)`

Inline information should be forrmatted:
`logrus.Infof("the expected value '%v' arrived as '%v'", expected, actual)`

And errors reporting inline information should be formatted:
`logrus.Errorf("the expected value '%v did not compute: %v", value, err)`